.github/workflows/lint-vizro-all.yml

```diff
@@ -0,0 +1,47 @@
+name: lint-vizro-all
+
+defaults:
+  run:
+    working-directory: vizro-core # but could be any folder
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches:
+      - "main"
+
+concurrency:
+  group: lint-${{ github.head_ref }}
+  cancel-in-progress: true
+
+env:
+  PYTHONUNBUFFERED: "1"
+  FORCE_COLOR: "1"
+
+jobs:
+  run:
+    name: Python ${{ matrix.python-version }} on Linux
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ["3.8"]
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: ${{ github.event_name == 'pull_request' && 2 || 0 }}
+
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Hatch
+        run: pip install --upgrade hatch
+
+      - name: List dependencies
+        run: hatch run all.py${{ matrix.python-version }}:pip freeze
+
+      - name: Lint
+        run: hatch run all.py${{ matrix.python-version }}:lint
```

Comment:
Since linting just requires `pre-commit` actually we should probably make a new `deatched=True` Hatch environment `lint` for it that doesn't include all of the dev dependencies. This way we could just do `hatch run lint:lint` on CI and it would be significantly faster. At the moment we install all the dependencies for `all.py3.8` and then run `pre-commit` that uses only one of those dependencies which is a bit silly.

Not a big deal though, so feel free to do in a separate PR another time if you prefer.

Reply:
Hm good idea, let me see if I can incorporate that quickly still

Reply:
On the other hand, `hatch run lint` is really used a lot, so would prefer not to have `hatch run lint:lint`. Or do you mean to keep `pre-commit` in the default environment as well?

Reply:
Let's revisit in another PR if we still want to do changes

Reply:
@maxschulz-COL yes, you're right - I originally put `lint` in the `default` environment precisely so that we don't have to write `hatch run lint:lint`. But probably the right way to do it is to make the `hatch run lint:lint` the "correct" command with a detached `lint` environment and then put a script `"lint" = ["hatch run lint:lint"]` in the default environment to just act as a shortcut so you can still do `hatch run lint`.